### DISCUSSION TITLE
Support ClickHouse as a SkyWalking storage option

### DISCUSSION THREAD

**Discussion by forest003** [apache/skywalking#9011] [Q&A] [answered]
Clickhouse is currently supported in storage. Will it be supported later

**Comment by wu-sheng** [comment, marked as answer]
AFAIK, no, we wouldn't. At least no one is working on that.

**Reply by cxntsh** [reply]
Hello, Mr. Wu Sheng. We have completely implemented the Clickhouse adaptation internally. After our optimization and testing, its performance is far better than that of other relational databases, and its memory will be lower than that of ES.

**Reply by wu-sheng** [reply]
As an open-source project, we welcome everyone to expand the ecosystem to bring more options to the community.
If you want to open-source the implementation, I would suggest you open and distribute another binary package for the end-users.
I am looking forward to you could share the progress with the community.

**Reply by cxntsh** [reply]
Thank you Mr. Wu Sheng for his reply. We are a commercial IT monitoring manufacturer and are currently preparing to release a commercial version of APM based on Skywalking. For giving back to the community, we have a very strong will to do this. We have met a lot of customer needs in the first-tier market, which is very meaningful. But I have a concern that we lack experience in communicating about the rules of the community, including how to submit code. I will keep in touch with the community.

**Reply by wu-sheng** [reply]
It is totally fine if this is in commercial, that is Apache 2.0 granted.

If you want this open to the public, it is good news for the community. Then you could open it on your own side, but declare what you did on the top of the official release. That is Apache 2.0 is required. 
This is new to me, as well as to the community, so, I would say this may not be a perfect time to discuss submitting/merging this part of codes into the main repo directly. As you mentioned, `lack experience in communicating about the rules of the community` could cause confusion and concerns on both our sides.

**Reply by sglztc** [reply]
Hello cxntsh, does skywalking support clickhouse storage now?

**Reply by wu-sheng** [reply]
AFAIK, it exists, but only in the commercial offering. The upstream doesn't have that.
